integration/setup/teardown.go

```diff
@@ -9,50 +9,36 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/e2e-harness/pkg/framework"
-	awsclient "github.com/giantswarm/e2eclients/aws"
 	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/aws-operator/integration/env"
 )
 
-func HostPeerVPC(c *awsclient.Client, g *framework.Guest, h *framework.Host) error {
-	log.Printf("Deleting Host Peer VPC stack")
-
-	_, err := c.CloudFormation.DeleteStack(&cloudformation.DeleteStackInput{
-		StackName: aws.String("host-peer-" + env.ClusterID()),
-	})
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
-	return nil
-}
-
-func Teardown(c *awsclient.Client, g *framework.Guest, h *framework.Host) error {
+func teardown(config Config) error {
 	var err error
 	var errors []error
 
 	{
-		err = framework.HelmCmd(fmt.Sprintf("delete %s-aws-operator --purge", h.TargetNamespace()))
+		err = framework.HelmCmd(fmt.Sprintf("delete %s-aws-operator --purge", config.Host.TargetNamespace()))
```

Comment:
As an intermediate step this is already a good move. I just think that we should have proper objects and that these things here should be provided by the pointer receiver and we simply call methods on the instantiated object. But we can iterate on this.

Reply:
ATM it just exposes a single method, so I don't have a strong feeling there. But if we decide to do so the transition will be rather straight forward.